cardano-node-chairman/src/Testnet/Byron.hs

```diff
@@ -70,22 +73,59 @@ defaultTestnetOptions = TestnetOptions
   , securityParam = 10
   , nPoorAddresses = 128
   , totalBalance = 8000000000000000
+  , enableP2P = False
   }
 
 replaceNodeLog :: Int -> String -> String
 replaceNodeLog n s = T.unpack (T.replace "logs/node-0.log" replacement (T.pack s))
   where replacement = T.pack ("logs/node-" <> show @Int n <> ".log")
 
 -- | Rewrite a line in the configuration file
-rewriteConfiguration :: Int -> String -> String
-rewriteConfiguration _ "TraceBlockchainTime: False" = "TraceBlockchainTime: True"
-rewriteConfiguration n s | "logs/node-0.log" `L.isInfixOf` s = replaceNodeLog n s
-rewriteConfiguration _ s = s
+rewriteConfiguration :: Bool -> Int -> String -> String
+rewriteConfiguration _ _ "TraceBlockchainTime: False"          = "TraceBlockchainTime: True"
+rewriteConfiguration _ n s | "logs/node-0.log" `L.isInfixOf` s = replaceNodeLog n s
+rewriteConfiguration True _ "EnableP2P: False"                 = "EnableP2P: True"
+rewriteConfiguration False _ "EnableP2P: True"                 = "EnableP2P: False"
+rewriteConfiguration _ _ s                                     = s
 
 rewriteParams :: TestnetOptions -> Value -> Value
 rewriteParams testnetOptions = rewriteObject
   $ HM.insert "slotDuration" (J.toJSON @String (show @Int (slotDuration testnetOptions)))
 
+mkTopologyConfig :: Int -> Int -> [Int] -> Bool -> ByteString
+mkTopologyConfig i numBftNodes allPorts False =
```

Comment:
@newhoggy I have a comment outside of this PR, I think writing `json` object in this way is a bit strange.  I think it would be nicer to have a `ToJSON` instance for topology files and use the data structure and turn it into json.

Reply:
The JSON instance for the Topology format already exists. I also found this style very weird, especially when there are template files that we can read, parse, manipulate and turn into json again, much more simpler!

Reply:
If there's already an accessible type, then by all means 😁

Reply:
Do we have a tests that checks that `ToJSON` and `FromJSON` are compatible?

Reply:
Yes: https://github.com/input-output-hk/cardano-node/blob/p2p-master/cardano-node/test/Test/Cardano/Node/Json.hs#L51